### ISSUE TITLE
Always have two months of time entries fetched

### ISSUE THREAD

**Issue by amulware** [toggl/mobileapp#1332]
We want to fetch these as soon as the user logs in. We still have to figure out what endpoints to use. Ideally, we can fetch the TEs by start date, not by AT date.

This issue might be relevant: https://github.com/toggl/backend/issues/6110

Note that if the user has not synced for too long so that we would reset the time entry since date, we should again fetch the full two months from scratch.

**Comment by amulware** [comment]
The above issue is indeed relevant. The endpoint created there is exactly what we'll use. Note that we should still always do the since call/regular call without since that data will be cached and faster.

We can display that data and persist it right away and then call the new endpoint to get the last two months of data. **This only has to happen if we called the regular endpoint without since date.**

**Comment by amulware** [comment]
This'll require the new endpoint to be implemented, ideally in its own PR with good integration tests.

**Comment by simonrozsival** [comment]
I am quite confused - we have three cases:
- get "all" (last 10 days of activity - compared by `at`, not by start) - "regular call without since"
- get all since some date
- get last two months

Which case should be used when? Could you summarize it somehow?

Right now the logic is:
```
if (since.HasValue && since.Value is less than two months old):
    get all since
else:
    get all without since (last 10 days)
```

**Comment by simonrozsival** [comment]
ping @amulware just in case you missed my question, ignore if it is already in your queue

**Comment by amulware** [comment]
_in principle_ we should do this:
```
if (since.HasValue && since.Value is less than two months old):
    get all since
else:
    get all in last 8 weeks
```

However, the 10-day endpoint is potentially a lot faster, since that data should be cached or something, so we should do:

```
if (since.HasValue && since.Value is less than two months old):
    get all since
else:
    get all without since (last 10 days)
    get all in last 8 weeks in parallel/after
```

Now the question is: Does this really get us any speedup anywhere? Maybe we should just remove the 10-day thing?

Certainly on login right now, we wait before showing UI until everything is pulled.

Ideally, we would maybe show UI after pulling 10 days, and then load 8 weeks in the background while the user can already show the app. Though I'm not sure how tricky that is. Nor do I know how much time it will save, since we're not measuring right now (can you try just curling locally to see if there is a significant difference in time? especially on the first request (repeated ones might be cached)).

In the case of already being logged in and the user coming back after two+ months, do we care much how long the initial sync takes? Ideally, we show data as fast as possible of course...

It comes down to the time difference between the endpoints. To we have any idea? Can we test it?

**Comment by simonrozsival** [comment]
Data for my personal account (where I have data for each week day):

| function | # | ms | kB | 
| -------- | - | -- | -- |
| get all without since (last 10 days) | 43 | 408 | 15.91 |
| get last two months | 190 | 586 | 67.88 |

I think that if we display some nice loading msg (and let user play with the spider), he/she can wait a few hundred ms so we can show him/her more relevant data.

**Comment by amulware** [comment]
That's not a lot of time entries compared to some users, but it also doesn't seem like a big deal for the time being. I'm ok just replacing the 10-day thing with the 8-week one for now.

We can optimise later.

**Comment by simonrozsival** [comment]
This issue was resoved by #2284 and #2357. I am closing this issue manually now (there was an incorrect description, so it was not closed automatically).
